src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/takemanagement/view/RecordableFragment.kt

```diff
@@ -0,0 +1,202 @@
+            recordableViewModel.deleteTake(it.take)
+        }
+        addEventHandler(EditTakeEvent.EDIT_TAKE) {
+            recordableViewModel.editTake(it)
+        }
+    }
+
+    private fun createSnackBar(pane: Pane) {
+        // TODO: This doesn't actually handle anything correctly. Need to know whether the user
+        // TODO... hasn't selected an editor or recorder and respond appropriately.
+        val snackBar = JFXSnackbar(pane)
+        recordableViewModel.snackBarObservable.subscribe {
+            snackBar.enqueue(
+                JFXSnackbar.SnackbarEvent(
+                    messages["noRecorder"],
+                    messages["addPlugin"].toUpperCase(),
+                    5000,
+                    false,
+                    EventHandler {
+                        audioPluginViewModel.addPlugin(true, false)
+                    })
+            )
+        }
+    }
+
+    private fun createAudioPluginProgressDialog() {
+        // Plugin active cover
+        val dialog = progressdialog {
+            root.addClass(AppStyles.progressDialog)
+            recordableViewModel.contextProperty.toObservable().subscribe { newContext ->
+                when (newContext) {
+                    TakeContext.RECORD -> graphic = AppStyles.recordIcon("60px")
+                    TakeContext.EDIT_TAKES -> graphic = AppStyles.editIcon("60px")
+                    else -> {}
+                }
+            }
+        }
+        recordableViewModel.showPluginActiveProperty.onChange {
+            Platform.runLater {
+                if (it) dialog.open() else dialog.close()
+            }
+        }
+    }
+
+    private fun Node.bindVisibleToDraggingNodeProperty() {
+        visibleProperty().bind(draggingNodeProperty.booleanBinding { it != null })
+    }
+
+    private fun startDrag(evt: StartDragEvent) {
+        if (evt.take != recordableViewModel.selectedTakeProperty.value) {
+            draggingNodeProperty.set(evt.draggingNode)
+            animateDrag(evt.mouseEvent)
+        }
+    }
+
+    private fun animateDrag(evt: AnimateDragEvent) {
+        animateDrag(evt.mouseEvent)
+    }
+
+    private fun animateDrag(evt: MouseEvent) {
+        if (draggingNodeProperty.value != null) {
+            val widthOffset = 348
+            val heightOffset = 200
+            dragContainer.toFront()
+            dragContainer.relocate(evt.sceneX - widthOffset, evt.sceneY - heightOffset)
+        }
+    }
+
+    private fun onDraggedToTarget(take: Take) {
+        recordableViewModel.selectTake(take)
+    }
+
+    private fun completeDrag(evt: CompleteDragEvent) {
+        if (dragTargetTop.contains(dragTargetTop.sceneToLocal(evt.mouseEvent.sceneX, evt.mouseEvent.sceneY))) {
+            onDraggedToTarget(evt.take)
+            draggingNodeProperty.set(null)
+        } else cancelDrag(evt.onCancel)
+    }
+
+    private fun cancelDrag(onCancel: () -> Unit) {
+        draggingNodeProperty.set(null)
+        onCancel()
+    }
```

Comment:
Consider reworking the responsibilities of these last two or three functions.  I might inline `cancelDrag`, and then refactor the redundant `draggingNodeProperty.set(null)` outside the resulting `if/else`.

Reply:
Yup, done